homeassistant/components/smartthings/fan.py

```diff
@@ -41,19 +41,50 @@ async def async_setup_entry(
 
 def get_capabilities(capabilities: Sequence[str]) -> Sequence[str] | None:
     """Return all capabilities supported if minimum required are present."""
-    supported = [Capability.switch, Capability.fan_speed]
-    # Must have switch and fan_speed
-    if all(capability in capabilities for capability in supported):
-        return supported
-    return None
+
```

Comment:
Its how the component creates entities from a device capability
https://github.com/home-assistant/core/blob/dev/homeassistant/components/smartthings/__init__.py#L311

If goes through platforms and calls get_capabilities then the entity takes those capabilities and you check if the next platform will take the remaining capabilities and so on.

Reply:
The get_capabilities call returns a list of capabilities for this entity. If the list is none then it continues to the next platform
https://github.com/home-assistant/core/blob/dev/homeassistant/components/smartthings/__init__.py#L312

Reply:
I actually made a custom component to make this work because I had just reserved that this integration is stale and @andrewsayre will not review given that #31230 is this issue and has been stale for years now and it took me an hour. I saw you had something merged and decided to try get a review and merge going into official integration

Reply:
I'm glad you opened this PR instead of just continuing with your custom component! It's better if people open PR's here so we can fix the issues.

What I meant was that on line 47 it says `If none of the optional capabilities are supported then error` but there are no error as it just continues so maybe we should change the comment to `If none of the optional capabilities are supported then return and don't setup platform` or something similar.

Reply:
Fixed the comment up

Reply:
Sorry to be a pest @gjohansson-ST but bump